src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -1950,11 +1950,6 @@ private async Task<Organization> GetOrgById(Guid id)
 
     private static void ValidatePlan(Models.StaticStore.Plan plan, int additionalSeats, string productType)
     {
-        if (plan is not { LegacyYear: null })
-        {
-            throw new BadRequestException($"Invalid {productType} plan selected.");
-        }
-
```

Comment:
This method is used (via `ValidatePasswordManagerPlan`) to prevent a user from signing up or upgrading to a legacy plan - which we still want to enforce.

Given that you can only signup/upgrade to PM, and SM is still an add-on, the quick solution is to move this `if` block into `ValidatePasswordManagerPlan` so that we retain the check on signup/upgrade but not on adding SM.

This validation logic is a mess and should be refactored before we allow separate subscriptions to PM and/or SM.

Separately - this pattern matching hides a `null` check which we should keep. So after moving it, add back a null check here.

Reply:
Good catch, I updated `ValidatePasswordManagerPlan` to contain this check and update the validation method to ensure `plan` isn't `null`